src/havok/havok.jl

```diff
@@ -0,0 +1,165 @@
+        reg_method=reg_method,      # Could be a function or default method
+        interceptQ=interceptQ,      # Bias fit boolean flag
+        λ=λ,selectQ=selectQ)        # Params of default regression methods
+
+    #=== INTERMITTENTLT FORCED MODEL ===#
+
+    # From ControlSystems ⨰ = A x + B u; y = C x + D u
+    sys = intermittently_forced_state_space_system(β)
+
+    #=== SOLVE MODEL ===#
+
+    # HAVOKsim with whole time trajectory
+    sol = sim(sys, range, Embedding, dt)
+
+    # Create and return HAVOKmodel object
+    return HAVOKmodel(timeseries, d, q, r, dt, Embedding, dX, β, sys, sol)
+end
+
+
+function fit(Embedding::DelayEmbedding, dt::AbstractFloat;
+    der_method="CentralDifferenceFourthOrder",  # default method from DOI: 10.1038/s41467-017-00030-8
+    reg_method="SequentialLeastSquares",        # default method from DOI: 10.1038/s41467-017-00030-8
+    interceptQ=false,λ=0,selectQ=MinBIC())      # params of solvers and bias fit boolean flag
+
+    @unpack Eigenseries = Embedding
+    r = size(Eigenseries,2)
+
+    #=== SPARSE LINEAR REGRESSION ===#
+
+    # Build design matrix and its numerical derivative approximation
+    range, dX = NumericalDifferentiation(Eigenseries, dt; der_method=der_method)
+    X = Eigenseries[range,:]
+
+    norms = [norm(X[:,i]) for i in 1:r]     # columns norms
+    X = X./norms'   # broadcasts norms across respective columns => columns normalized
+
+    println("Using $der_method the numerical derivative ∈ $(size(dX)) was computed.")
+
+    # β fit in dX = X * β
+    β = RegressionSolve(X,dX;reg_method=reg_method,interceptQ=interceptQ,λ=λ,selectQ=selectQ)
+
+    println("Using $reg_method the regression coefficients ∈ $(size(β)) were computed.")
+
+    println("Quality of fit measured with pearson correlations for each derivative:")
+    for i in 1:r
+        println("$i: $(cor(dX[:,i],X*β[1+(end-r):end,i]))")
+    end
+
+    # Brunton normalizes the coefficients?!, maybe because the
+    # elimination of the entries in V where derivative didnt exist
+    # de-normalizes V but why apply the normalization on β?
+    length(β)==r*r ? (β = β./norms) : (β = β[1+(end-r):end,:]./norms) # Broadcasts across rows
+
+    return β, dX, range
+end
+
+
+function intermittently_forced_state_space_system(β::Array{<:AbstractFloat,2})
+    # Remember that y=X*β therefore (y[i,:])'=β'*(X[i,:])'
+    s, r = size(β)
+    βview = view(β, 1+(s-r):s, 1:s)' # transposed and intercept_col dropped
+
+    # Define the matrices state A, input B , output C, and feedforward D.
+    A = βview[1:r-1,1:r-1]
+    B = βview[1:r-1,r]          # forcing
+    C = Diagonal(ones(r-1))
+    D = zeros(size(B))
+
+    println("A model of the form ⨰ = A x + B u; y = C x + D u is proposed.\n")
+
+    # Define sys from ControlSystems ⨰ = A x + B u; y = C x + D u
+    return ss(A,B,C,D)
+end
+
+
+function sim(sys::StateSpace{R,Array{R,2}}, range::UnitRange{Int}, Embedding::DelayEmbedding{R}, dt::R) where R <: AbstractFloat
```

Comment:
This should be handled differently. But this is more of a comment for myself, since this requires a `DDE`system to be created from the result.